src/esp/sensor/CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ add_library(
 target_link_libraries(
   sensor
   PUBLIC core gfx scene
+  PRIVATE sim
```

Comment:
Why private?

Reply:
It was set (in different ways) to help me better understand why the CI fails.
It is a sad story. I was taking PTO on Mon. but I spent whole day trying to solve the CI failure, which is still there.

I found the `ResourceMangerTest` cannot be compiled. It complains `libsensor.a` is missing some definitions from here and there, even though it does not need anything from `sensor`.